src/js/pwm.js

```diff
@@ -97,77 +89,84 @@ function PWM(pin, options, callback) {
   });
 }
 
-// pwm.setPeriod(period[, callback])
-PWM.prototype.setPeriod = function (period, callback) {
-  var self = this;
-  // Check arguments.
+PWM.prototype._validatePeriod = function(period) {
   if (!util.isNumber(period)) {
-    throw new TypeError('Bad arguments - period');
+    throw new TypeError('Period is not a number(' + typeof(period) + ')');
+  } else if (period < 0) {
```

Comment:
@tadziopazur Okay, it was due to code readability. There is no problem in operation. Thank you for your answer.